### PR TITLE
🧹 use bom for vuln scanning

### PR DESCRIPTION
**Pull request by chris-rock**
depends on https://github.com/mondoohq/cnquery/pull/3533

This PR refactors the data gathering and uses cnquery bom data gathering instead. This simplifies the code and reduces the different ways we gather data.

### CODE REVIEW

File: apps/cnspec/cmd/vuln.go [hunk 1]
```diff
@@ -51,139 +46,91 @@ var vulnCmd = &cobra.Command{
 }
 
 var vulnCmdRun = func(cmd *cobra.Command, runtime *providers.Runtime, cliRes *plugin.ParseCLIRes) {
-	conf, err := getCobraScanConfig(cmd, runtime, cliRes)
-	// conf := cnquery_app.ParseShellConfig(cmd, cliRes)
+	log.Info().Msg("This command is experimental. Please report any issues to https://github.com/mondoohq/cnquery.")
```
Comment (czunker) [inline apps/cnspec/cmd/vuln.go]: nit: `vuln` has already been present for some time now, and now we mark it experimental?

File: apps/cnspec/cmd/vuln.go [hunk 2]
```diff
@@ -51,139 +46,91 @@ var vulnCmd = &cobra.Command{
...
+	if err != nil {
+		log.Fatal().Err(err).Msg("failed to parse report collection")
 	}
 
-	if value == nil || value.Data == nil {
-		log.Error().Msg("could not load packages data\n\n")
-		return
+	ctx := cmd.Context()
+	upstreamConf := conf.runtime.UpstreamConfig
+	client, err := upstreamConf.InitClient(ctx)
```
Comment (czunker) [inline apps/cnspec/cmd/vuln.go:90]: This panics, when there is no upstream config:
```
panic({0x1bb9720?, 0x348f1f0?})
	/home/linuxbrew/.linuxbrew/Cellar/go/1.22.0/libexec/src/runtime/panic.go:770 +0x132
go.mondoo.com/cnquery/v10/providers-sdk/v1/upstream.(*UpstreamConfig).InitClient(0x0, {0x259cc98, 0x356fe20})
	/home/christian/go/pkg/mod/go.mondoo.com/cnquery/v10@v10.6.2-0.20240311174838-3eb7077113c2/providers-sdk/v1/upstream/upstream.go:59 +0x35
go.mondoo.com/cnspec/v10/apps/cnspec/cmd.init.func34(0xc00091f508, 0xc00111d400, 0xc00117c540)
	/home/christian/workspace/mondoo/github.com/cnspec/apps/cnspec/cmd/vuln.go:84 +0x3bb
```

File: apps/cnspec/cmd/vuln.go [hunk 3]
```diff
@@ -51,139 +46,91 @@ var vulnCmd = &cobra.Command{
...
+	// TODO: support origin, add cpes and purls
+	for i := range bom.Packages {
+		pkg := bom.Packages[i]
+		req.Packages = append(req.Packages, &mvd.Package{
+			Name:    pkg.Name,
+			Version: pkg.Version,
+			//Arch:    pkg.Arch,
+			//Purl:    pkg.Purl,
+			//Cpes:    pkg.Cpes,
+			//Format:  pkg.Format,
```
Comment (czunker) [inline apps/cnspec/cmd/vuln.go]: We need the format for some of the app vulns, to distinguish between, e.g., the mac and the windows package.

File: apps/cnspec/cmd/vuln.go (hunk 1 above)
Comment (chris-rock) [reply apps/cnspec/cmd/vuln.go]: I am going to remove that flag here

File: apps/cnspec/cmd/vuln.go (hunk 3 above)
Comment (chris-rock) [reply apps/cnspec/cmd/vuln.go]: origin and format is added now

**Review by czunker — APPROVED** [review]
LGTM:
```
Target:     debian:buster-CVE-2023-38408@b9bb5a146596
=======================================================

┌─ Advisories ─────────────────────────┐
│ Critical: ███ 14.3%                  │
│ High:     ████████████ 42.9%         │
│ Medium:   ███ 14.3%                  │
│ Low:      ███ 14.3%                  │
│ None:     ███ 14.3%                  │
└──────────────────────────────────────┘

┌─ Packages ───────────────────────────┐
│ Total:    ███████████████████ 109    │
│ Critical: ███ 1                      │
│ High:     ███████████████████ 5      │
│ Medium:   0                          │
│ Low:      ███ 1                      │
└──────────────────────────────────────┘

  ■   SCORE  PACKAGE         INSTALLED               FIXED                   AVAILABLE               ADVISORY    
  ■   9.8    openssh-client  1:7.9p1-10+deb10u2      1:7.9p1-10+deb10u4      1:7.9p1-10+deb10u4                  
  ├─  9.8    openssh-client  1:7.9p1-10+deb10u2      1:7.9p1-10+deb10u3      1:7.9p1-10+deb10u4      DLA-3532-1  
  ╰─  7      openssh-client  1:7.9p1-10+deb10u2      1:7.9p1-10+deb10u4      1:7.9p1-10+deb10u4      DLA-3694-1  
  ■   8.8    libncursesw6    6.1+20181013-2+deb10u4  6.1+20181013-2+deb10u5  6.1+20181013-2+deb10u5              
  ╰─  8.8    libncursesw6    6.1+20181013-2+deb10u4  6.1+20181013-2+deb10u5  6.1+20181013-2+deb10u5  DLA-3682-1  
  ■   8.8    libtinfo6       6.1+20181013-2+deb10u4  6.1+20181013-2+deb10u5  6.1+20181013-2+deb10u5              
  ╰─  8.8    libtinfo6       6.1+20181013-2+deb10u4  6.1+20181013-2+deb10u5  6.1+20181013-2+deb10u5  DLA-3682-1  
  ■   8.8    ncurses-base    6.1+20181013-2+deb10u4  6.1+20181013-2+deb10u5  6.1+20181013-2+deb10u5              
  ╰─  8.8    ncurses-base    6.1+20181013-2+deb10u4  6.1+20181013-2+deb10u5  6.1+20181013-2+deb10u5  DLA-3682-1  
  ■   8.8    ncurses-bin     6.1+20181013-2+deb10u4  6.1+20181013-2+deb10u5  6.1+20181013-2+deb10u5              
  ╰─  8.8    ncurses-bin     6.1+20181013-2+deb10u4  6.1+20181013-2+deb10u5  6.1+20181013-2+deb10u5  DLA-3682-1  
  ■   7.5    libgnutls30     3.6.7-4+deb10u10        3.6.7-4+deb10u12        3.6.7-4+deb10u12                    
  ├─  7.5    libgnutls30     3.6.7-4+deb10u10        3.6.7-4+deb10u12        3.6.7-4+deb10u12        DLA-3740-1  
  ╰─  5.9    libgnutls30     3.6.7-4+deb10u10        3.6.7-4+deb10u11        3.6.7-4+deb10u12        DLA-3660-1  
  ■   3.3    tar             1.30+dfsg-6             1.30+dfsg-6+deb10u1     1.30+dfsg-6+deb10u1                 
  ╰─  3.3    tar             1.30+dfsg-6             1.30+dfsg-6+deb10u1     1.30+dfsg-6+deb10u1     DLA-3755-1  
  ■   0      tzdata          2021a-0+deb10u11        2021a-0+deb10u12        2021a-0+deb10u12                    
  ╰─  0      tzdata          2021a-0+deb10u11        2021a-0+deb10u12        2021a-0+deb10u12        DLA-3684-1  
```
Thanks @chris-rock

**Pull request merged**